manticore/core/cpu/abstractcpu.py

```diff
@@ -477,6 +478,74 @@ def read_bytes(self, where, size):
             result.append(Operators.CHR(self.read_int( where+i, 8)))
         return result
 
+    def read_string(self, where, max_length=None):
+        '''
+        Read a NUL-terminated concrete buffer from memory.
+
+        :param int where: Address to read string from
+        :param int max_length:
+            The size in bytes to cap the string at, or None [default] for no
+            limit.
+        :return: string read
+        :rtype: str
+        '''
+        s = StringIO.StringIO()
+        while True:
+            c = self.read_int(where, 8)
+            if c == 0:
+                break
+            if max_length is not None:
+                if max_length == 0:
+                    break
+                max_length = max_length - 1
+            s.write(Operators.CHR(c))
+            where += 1
+        return s.getvalue()
+
+    def push_bytes(self, data):
+        '''
+        Write `data` to the stack and decrement the stack pointer accordingly.
+
+        :param str data: Data to write
+        '''
+        self.STACK -= len(data)
```

Comment:
So aliases are passed to RegisterFile constructor and those may not contain 'STACK' as it is now. If we want STACK to be part of RegisterFile I say we should assert that an specific alias for it is provided. This is at RegisterFile constructor or AbstractCpu I guess.  Same for 'PC'. No?